src/app/ReadHandler.cpp

```diff
@@ -652,6 +719,52 @@ CHIP_ERROR ReadHandler::ProcessSubscribeRequest(System::PacketBufferHandle && aP
 
     mExchangeCtx->WillSendMessage();
 
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    auto * subscriptionResumptionStorage = InteractionModelEngine::GetInstance()->GetSubscriptionResumptionStorage();
+    if (!subscriptionResumptionStorage)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    SubscriptionResumptionStorage::SubscriptionInfo subscriptionInfo = { .mNodeId         = GetInitiatorNodeId(),
+                                                                         .mFabricIndex    = GetAccessingFabricIndex(),
+                                                                         .mSubscriptionId = mSubscriptionId,
+                                                                         .mMinInterval    = mMinIntervalFloorSeconds,
+                                                                         .mMaxInterval    = mMaxInterval,
+                                                                         .mFabricFiltered = IsFabricFiltered() };
+    ObjectList<AttributePathParams> * attributePath                  = mpAttributePathList;
+    size_t attributePathCount                                        = 0;
+    while (attributePath)
+    {
+        attributePathCount++;
+        attributePath = attributePath->mpNext;
+    }
+    attributePath = mpAttributePathList;
+    subscriptionInfo.mAttributePaths.Calloc(attributePathCount);
```

Comment:
Can this allocation fail? How is that handled?

Reply:
I've added nullptr checks